gourmet/backends/db.py

```diff
@@ -1158,10 +1157,8 @@ def validate_recdic (self, recdic):
                 import traceback
                 traceback.print_stack()
         for k,v in list(recdic.items()):
-            try:
-                recdic[k]=str(v.strip())
-            except:
-                pass
+            if isinstance(v, str):
+                recdic[k] = v.strip()
```

Comment:
We're allowed to update dictionaries in place now. So this works

```python
foo = {1: 2, 3: 4, 5: 6}
for k, v in foo.items():
        foo[k] = v+1
print(foo)
```

we can get rid of the `list` conversion.

Reply:
I wasn't aware of a change in that space. The docs still say that you mustn't add or delete entries while iterating over a dict view; I think technically that means replacing existing entries is OK, but I'd rather play it safe and leave the `list()` call. The extra memory use should be tiny and momentary.